PWGHF/D2H/Macros/efficiency.py

```diff
@@ -127,13 +129,29 @@ def compute_efficiency(
             h_rec = h_rec.ProjectionY()
 
     if is_gen_2d:
-        if rapidity_cut is not None:
-            h_gen.GetYaxis().SetRangeUser(-1.0 * rapidity_cut + epsilon, rapidity_cut - epsilon)
         if axis_rapidity == 'Y':
+            if rapidity_cut is not None:
+                h_gen.GetYaxis().SetRangeUser(-1.0 * rapidity_cut + epsilon, rapidity_cut - epsilon)
             h_gen = h_gen.ProjectionX()
         else:
+            if rapidity_cut is not None:
+                h_gen.GetXaxis().SetRangeUser(-1.0 * rapidity_cut + epsilon, rapidity_cut - epsilon)
             h_gen = h_gen.ProjectionY()
 
+    # rebin histograms, if enabled
+    if pt_bins_limits is not None:
+        n_bins = len(pt_bins_limits) - 1
+
+        h_rec = h_rec.Rebin(n_bins, 'hRec', pt_bins_limits)
+        # using Scale causes weights-related complications in TEfficiency
+        # trick: we SetBinContent + Sumw2 (gives same result as w/o rebinning)
+        h_rec.SetBinContent(1, h_rec.GetBinContent(1) / h_rec.GetBinWidth(1))
```

Comment:
probably I'm overlooking something, but why are you touching here only the bin 1? You have just rebinned your `h_rec` and in principle you'd like to normalize all the bins, right? The same for the `h_gen` below